pkg/endpoint/bpf.go

```diff
@@ -342,16 +342,29 @@ func (e *Endpoint) hashHeaderfile(prefix string) (string, error) {
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
 	hashWriter := md5.New()
-	for scanner.Scan() {
-		line := scanner.Text()
-		if strings.HasPrefix(line, "#") {
-			io.WriteString(hashWriter, line)
+	reader := bufio.NewReader(file)
+	firstFragmentOfLine := true
+	lineToHash := false
+	for {
+		fragment, isPrefix, err := reader.ReadLine()
+		if err != nil {
+			if err == io.EOF {
```

Comment:
I assume because this is unstructured it never returns `ErrUnexpectedEOF`?

Reply:
I tested that it returns `io.EOF`.